src/components/text-input/TextInput.js

```diff
@@ -69,10 +75,15 @@ const TextInput = ({
                 showOptionalText={!hideOptionalLabel && !isRequired}
                 text={label}
                 tooltip={labelTooltip}
+                htmlFor={textInputID}
```

Comment:
I don't think it's a good idea to do `<label for="something">...</label>` when you use label element as container. It's like trying to do implicit and explicit label association at the same time. Could you remove it?